slick/src/main/scala/slick/lifted/Query.scala

```diff
@@ -155,6 +155,11 @@ sealed abstract class Query[+E, U, C[_]] extends QueryBase[C[U]] { self =>
     new WrappingQuery[(G, Query[P, U, Seq]), (T, Query[P, U, Seq]), C](group, key.zip(value))
   }
 
+  /** Specify part of a select statement for update and marked for row level locking */
+  def forUpdate: Query[E, U, C] = {
```

Comment:
Although not all the backends support forUpdate in complex queries, some do. For example, MYSQL does http://stackoverflow.com/questions/18318681/nested-update-with-select-deadlock/18646713#18646713. By having it as a query language feature, we can support those more complex queries as well as the forUpdate syntax that specifies explicit columns, as mentioned above.